### PR TITLE
Strip more unused pod fields

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Review by adiprerepa — APPROVED** [review]
otherwise lgtm

File: pkg/kube/util.go [hunk 1]
```diff
@@ -350,9 +350,16 @@ func StripPodUnusedFields(obj any) (any, error) {
 				})
 			}
 		}
-		pod.Spec.Containers = containers
-		pod.Spec.InitContainers = nil
-		pod.Spec.Volumes = nil
+		oldSpec := pod.Spec
```
[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pkg/kube/util.go:353]: just to reduce writing of pod.spec

[the remark this answers is not in the document]
Comment (ramaraochavali) [reply pkg/kube/util.go:353]: How do we validate if we use a new field in pod spec in future code? Is there some way to ensure that it is taken care?

[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pkg/kube/util.go:353]: Without updating code here you will always get an empty or default value

**Pull request merged**